runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -5777,7 +5778,15 @@ class INTERPRETER_CLASS
 				omrthread_monitor_t monitor = getMonitorForWait(_currentThread, waitObject);
 				monitor->count = _currentThread->currentContinuation->waitingMonitorEnterCount;
 				_currentThread->currentContinuation->waitingMonitorEnterCount = 0;
-				returnVoidFromINL(REGISTER_ARGS, 4);
+				_currentThread->ownedMonitorCount -= 1;
+				if (J9VMJAVALANGTHREAD_DEADINTERRUPT(_currentThread, _currentThread->threadObject)) {
+					buildInternalNativeStackFrame(REGISTER_ARGS);
+					updateVMStruct(REGISTER_ARGS);
+					setCurrentException(_currentThread, J9VMCONSTANTPOOL_JAVALANGINTERRUPTEDEXCEPTION, NULL);
+					rc = GOTO_THROW_CURRENT_EXCEPTION;
```

Comment:
based on other examples in this file, it should be:
```suggestion
					updateVMStruct(REGISTER_ARGS);
					prepareForExceptionThrow(_currentThread);
					setCurrentException(_currentThread, J9VMCONSTANTPOOL_JAVALANGINTERRUPTEDEXCEPTION, NULL);
					VMStructHasBeenUpdated(REGISTER_ARGS);
					rc = GOTO_THROW_CURRENT_EXCEPTION;
```